### PR TITLE
Chat

### PR DESCRIPTION
**Pull request by kek**
* Upgrade to Phoenix 0.8.0.
* Add GitHub authentication.
* Demonstrates channels.
* Demonstrates workers and GenServer for keeping track of which users are on the chat.

To do:
* If the OAuth callback from GitHub contains something other than the normal successful login, the behavior is undefined. I think the chat will work but will display an empty username.
* Disconnected users will not be removed from the user list.

Demo at http://sheltered-shelf-2304.herokuapp.com/

With help from @kitofr @littlekid @martinsvalin @MikaelAmborn

### CODE REVIEW

File: lib/stockholm_elixir/endpoint.ex [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    json_decoder: Poison
+
+  plug Plug.MethodOverride
+  plug Plug.Head
+
+  plug Plug.Session,
+    store: :cookie,
+    key: "_stockholm_elixir_key",
+    signing_salt: "9HUmdyG3",
+    encryption_salt: "4L8wmQTR"
```
Comment (joakimk) [inline lib/stockholm_elixir/endpoint.ex]: Maybe this should be ENVs? Seems like a security issue.

File: web/router.ex [hunk 2]
```diff
@@ -4,16 +4,58 @@ defmodule StockholmElixir.Router do
...
+    assign(conn, :current_user, get_session(conn, :current_user))
+  end
+
+  defp assign_user_token(conn, _) do
+    username = get_session(conn, :current_user)["login"]
+    assign(conn, :user_token, make_token(username))
+  end
+
+  defp make_token(username) do
+    signing_key = System.get_env("SIGNING_KEY")
```
Comment (joakimk) [inline web/router.ex]: Maybe this could be pushed into config so we only read envs in one place.

**Comment by joakimk** [comment]
Merged and deployed, but haven't setup oauth stuff yet. Will look into later, no more time now.

File: lib/stockholm_elixir/endpoint.ex (hunk 1 above)
Comment (kek) [inline lib/stockholm_elixir/endpoint.ex]: It should definitely be.

**Comment by joakimk** [comment]
I've disabled the login link until there is config for it.

**Comment by kek** [comment]
Register it here: https://github.com/settings/applications. I could do it but it's probably best if the

**Comment by joakimk** [comment]
It's set up now.

**Comment by joakimk** [comment]
Some html sanitisation would probably be a good idea for this feature.

**Pull request merged**